app/src/main/java/com/example/steamapp/data/repository/UsersInfoRepositoryImpl.kt

```diff
@@ -0,0 +1,53 @@
+package com.example.steamapp.data.repository
+
+import com.example.steamapp.R
+import com.example.steamapp.data.api.SteamApi
+import com.example.steamapp.data.mapper.toDomain
+import com.example.steamapp.domain.model.UsersFriend
+import com.example.steamapp.domain.model.UsersInfo
+import com.example.steamapp.domain.repository.UserInfoRepository
+import com.example.steamapp.presentation.MySteamApplication
+
+class UsersInfoRepositoryImpl(private val steamApi: SteamApi) : UserInfoRepository {
+    override suspend fun getUsersInfoLst(): List<UsersInfo> {
+        val steamidStrLst = mutableListOf<String>()
+        val playerInfoLst = mutableListOf<UsersInfo>()
+
+        val lstOfFriends = getUserFriendsIds()
+
+        steamidStrLst.addAll(
+            lstOfFriends.map { it.steamId }
+        )
+
+        steamidStrLst.forEach {
+            val playersResp = getUserInfoById(it)
+            playerInfoLst.add(playersResp)
+        }
+        return playerInfoLst
+    }
```

Comment:
здесь проблема, ты будешь выполнять последовательно n запросов, нужно их параллелить через async/await

```suggestion
    override suspend fun getUsersInfoLst(): List<UsersInfo> = coroutineScope {
        val steamidStrLst = getUserFriendsIds().map { it.steamId }
        steamidStrLst
            .map { id ->
                async { getUserInfoById(id) }
             }
             .awaitAll()
    }
```